src/api/artifact/controller.go

```diff
@@ -48,6 +49,9 @@ type Controller interface {
 	List(ctx context.Context, query *q.Query, option *Option) (total int64, artifacts []*Artifact, err error)
 	// Get the artifact specified by ID, specify the properties returned with option
 	Get(ctx context.Context, id int64, option *Option) (artifact *Artifact, err error)
+	// Get the artifact specified by repository name and reference, the reference can be tag or digest,
+	// specify the properties returned with option
+	GetByReference(ctx context.Context, repository, reference string, option *Option) (artifact *Artifact, err error)
```

Comment:
WHY add this method? What's the use case?

Reply:
The name rather than ID is used in both Harbor management APIs and OCI registry APIs, such as:
Listing artifact API: `GET /api/v2.0/projects/{project_name}/repositories/{repository_name}/artifacts/{reference}`
Get manifest API: `GET /v2/{repository}/manifests/{reference}`
Provide this method can help implement such APIs easier.

Besides, the middlewares(immutable, content trust, vulnerability, etc.) can use this method directly to get the artifact information.

Without this method, every implementation should call:
1. RepositoryManager.List(repository_name) to get repository ID
2. TagManager.List(repository_name, tag_name) to get the artifact ID that the tag attached to
3. ArtifactManager.Get(artifact_id) to get the artifact information